src/plugins/ui_core_views/evaluation.ts

```diff
@@ -405,30 +956,124 @@ export class EvaluationPlugin extends UIPlugin {
     return [refFn, range, evalContext];
   }
 
+  private findCellsToCompute(mainRc: string, selfInclude: boolean = true): Iterable<string> {
+    const cellsToCompute = new Set<string>();
+    if (selfInclude) {
+      cellsToCompute.add(mainRc);
+    }
+    this.formulaDependencies.visitDeepReferences(mainRc, (rc: string) => cellsToCompute.add(rc));
```

Comment:
we should have visitDeepReferences return the create set of references instead of passing a second one to populate in the callback. We would avoid writing/reading a second (apparently) useless set.